core/arch/arm/kernel/kern.ld.S

```diff
@@ -61,6 +60,14 @@
 #define TEE_TEXT_VA_START       (TEE_RAM_VA_START + \
 					(TEE_LOAD_ADDR - TEE_RAM_START))
 
+/*
+ * Note:
+ * Clang 11 (ld.lld) generates non-relocatable references to symbols defined
+ * with ROUNDUP()/ROUNDDOWN() from <util.h>, which does not work with ASLR.
+ */
+#define LD_ROUNDUP(x, y)  (x) + (((x) % (y)) ? ((y) - ((x) % (y))) : 0)
```

Comment:
Out of curiosity, did you attempt to implement these two based on the builtin function `ALIGN()`?

Reply:
No I didn't, good suggestion. Updated.